frontend/src/app/core/services/user.service.ts

```diff
@@ -38,4 +41,12 @@ export class UserService {
     public getUserChallengesInfoByTags(id: number): Observable<IUserSolutionsGroupedBySkillLevel[]> {
         return this.httpService.getRequest<IUserSolutionsGroupedBySkillLevel[]>(`/users/${id}/user-challenges`);
     }
+
+    public getLeaderBoard(page?: IPageSettings): Observable<IUser[]> {
+        let httpParams = new HttpParams();
+
+        httpParams = setParams<IPageSettings>(httpParams, page);
+
+        return this.httpService.getRequest<IUser[]>(`${this.baseUrl}/leader-board`, httpParams);
```

Comment:
getLeaderboard(page?: IPageSettings): Observable<IUser[]> {
    const params = setParams(page);

    return this.http.get<IUser[]>(`${this.baseUrl}/leader-board`, { params });
  }
```suggestion
    public getLeaderBoard(page?: IPageSettings): Observable<IUser[]> {
        const params = setParams(page);
        
        return this.http.get<IUser[]>(`${this.baseUrl}/leader-board`, { params });
```

Reply:
This is not work in such way, in other places we use the same way of defining of httpParams